src/parachain/vaults.ts

```diff
@@ -516,6 +520,17 @@ export class DefaultVaultsAPI extends DefaultTransactionAPI implements VaultsAPI
         }
     }
 
+    async getVaultsWithRedeemableTokens(): Promise<Map<AccountId, Big>> {
+        try {
+            const vaults = await this.api.rpc.vaultRegistry.getVaultsWithRedeemableTokens();
+            return new Map(
+                vaults.map(([id, redeemableTokens]) => [id, new Big(satToBTC(this.unwrapCurrency(redeemableTokens).toString()))])
+            );
+        } catch (e) {
+            return Promise.reject("Did not find vault with issuable tokens");
```

Comment:
This is not minor. :) A major mistake.

Reply:
We've actually removed this error condition on the parachain recently, so there's no need for the try-catch anymore (it can be removed from `getVaultsWithIssuableTokens()` too).

Reply:
We are returning an empty array in case we don't find a vault but it might throw other error (like a network timeout). Then we should still have the try-catch here

Reply:
I don't think the other lib functions have try-catches for things like network errors. I think it should be up to the UI to handle errors like that. What should the lib do if there's a network error? The lib doesn't know who's calling it. Meanwhile, the UI can catch the error and display feedback to the user appropriately.